### PR TITLE
Add FreeBSD support

### PR DESCRIPTION
**Pull request by DentonGentry**

### CODE REVIEW

File: peercred_freebsd.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+func init() {
+	osGet = getFreeBSD
+}
+
+func getFreeBSD(c net.Conn) (*Creds, error) {
+	switch c := c.(type) {
+	case *net.UnixConn:
+		return getUnix(c)
+	case *net.TCPConn:
+		// TODO: use /proc tcp info for localhost connections like Windows?
```
Comment (bradfitz) [inline peercred_freebsd.go]: does FreeBSD have a /proc/net/tcp like Linux? I don't know how netstat works on BSD.

File: peercred_freebsd.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		}
+	})
+	if cerr != nil {
+		return nil, fmt.Errorf("raw.Control: %w", err)
+	}
+	if err != nil {
+		return nil, err
+	}
+	return &Creds{
+		pid: 0, // FreeBSD 13 adds a cr_pid field, can be used here.
```
Comment (bradfitz) [inline peercred_freebsd.go:52]: @tklauser, is this field on x/sys/unix?

**Review by bradfitz — APPROVED** [review]

File: peercred_freebsd.go (hunk 2 above)
Comment (tklauser) [reply peercred_freebsd.go:52]: No it's not (yet). Looks like it's defined as part of an `union` in `sys/ucred.h` (https://github.com/freebsd/freebsd-src/blob/de1aa3dab23c06fec962a14da3e7b4755c5880cf/sys/sys/ucred.h#L105) , so it wasn't generated

File: peercred_freebsd.go (hunk 1 above)
Comment (DentonGentry) [reply peercred_freebsd.go]: Updated to refer to sysctl.

**Comment by DentonGentry** [comment]
Updated /proc/net comment to indicate sysctl instead.

I cannot merge the PR myself as I don't have write access, but I think it is ready.

**Pull request merged**